[FIX] point_of_sale: delete silently in localDeleteCascade

pedrambiria opened the pull request:
Before this commit, cancelling an order would result in keeping the removed orderlines in the unlink command, causing issues in different cases when trying to sync a new order.

opw-4403100

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/192118.png)](https://mergebot.odoo.com/odoo/odoo/pull/192118)

pedrambiria commented:
@davidmonnom Hello,
Could you please review this pr?

davidmonnom reviewed (commented):
localDeleteCascade is used sometime to be able to delete the order from the server. With your change it will never do that.

```js
    removeOrder(order, removeFromServer = true) {
        if (this.isOpenOrderShareable() || removeFromServer) {
            if (typeof order.id === "number" && !order.finalized) {
                this.addPendingOrder([order.id], true);
            }
        }

        if (typeof order.id === "string" && order.finalized) {
            this.addPendingOrder([order.id]);
            return;
        }

        return this.data.localDeleteCascade(order, removeFromServer);
    }
```

I think in this case with need to mark the line as to delete in the commands obj of related models ?

pedrambiria commented:
@davidmonnom when we delete an order, we do not remove it from the server; instead, we mark it as canceled. Also, based on the function name `localDeleteCascade`, it seems the intention was to delete it only locally. If you believe we need to change this logic, I can modify it to remove the order from the server and also rename the function accordingly. What do you think?

davidmonnom commented:
> @davidmonnom when we delete an order, we do not remove it from the server; instead, we mark it as canceled. Also, based on the function name `localDeleteCascade`, it seems the intention was to delete it only locally. If you believe we need to change this logic, I can modify it to remove the order from the server and also rename the function accordingly. What do you think?

In this case, we'd have to change the check in this method to accept the local delete only when the order is cancel or not yet synced.

```js
        if (typeof record.id === "number" && !force) {
            console.info(
                `Record ID ${record.id} MODEL ${recordModel}. If you want to delete a record saved on the server, you need to pass the force parameter as true.`
            );
            return;
        }
```

pedrambiria commented:
@davidmonnom I modified the code. I replaced the `force` parameter with `removeFromServer` and set it to `false` when canceling an order. Additionally, I noticed that releasing a table did not update immediately, so I added a debounced `syncAllOrders` call in the `removeOrder` method to ensure timely updates.

davidmonnom commented on addons/point_of_sale/static/src/app/store/pos_store.js:
```diff
@@ -1083,6 +1085,7 @@ export class PosStore extends Reactive {
         if (this.isOpenOrderShareable() || removeFromServer) {
             if (typeof order.id === "number" && !order.finalized) {
                 this.addPendingOrder([order.id], true);
+                this.syncAllOrdersDebounced();
```
why debounced ? i think you can directly call syncallorders

pedrambiria replied:
@davidmonnom If you proceed with removing all the orders at once (e.g., when closing a session with open orders), it will trigger multiple calls unnecessarily, whereas only one call is needed.

davidmonnom reviewed (approved):
@robodoo r+
